migration-engine/connectors/sql-migration-connector/src/sql_migration.rs

```diff
@@ -458,6 +455,7 @@ pub(crate) enum SqlMigrationStep {
     CreateIndex {
         table_id: (Option<TableId>, TableId),
         index_index: usize,
+        from_drop_and_recreate: bool,
```

Comment:
Maybe add a comment about what this is / why it's needed?

Reply:
I'd rather not, because what it is is clear from context (we had other similar cases in SqlMigration), and the why would be far from where it's actually used and prone to getting out of sync. It's not part of the public API of the crate. The policy for these data structures is that if you can remove something and tests aren't failing, it's not necessary.